pkg/es/config/config.go

```diff
@@ -49,7 +49,7 @@ type Configuration struct {
 	AllowTokenFromContext bool           `mapstructure:"-"`
 	Sniffer               bool           `mapstructure:"sniffer"` // https://github.com/olivere/elastic/wiki/Sniffing
 	SnifferTLSEnabled     bool           `mapstructure:"sniffer_tls_enabled"`
-	MaxNumSpans           int            `mapstructure:"-"`                     // defines maximum number of spans to fetch from storage per query
+	MaxNumSpans           int            `mapstructure:"-"`                     // deprecated: use MaxDocCount instead. Defines maximum number of spans to fetch from storage per query
```

Comment:
This can be renamed to `MaxDocCount` and we need a test case when both flags are specified to show that the old flag works and also that the new one takes precedence.

Reply:
Okay, I misunderstood your comment in the Issue discussion, and thought you preferred to leave `MaxNumSpans` untouched while in a deprecated state. I've updated the diff to replace `MaxNumSpans` -> `MaxDocCount` in all places except for the user-facing flag.